velox/exec/ExchangeClient.cpp

```diff
@@ -130,38 +127,45 @@ ExchangeClient::next(uint32_t maxBytes, bool* atEnd, ContinueFuture* future) {
       return pages;
     }
 
-    requestSpec = pickSourcesToRequestLocked();
+    requestSpecs = pickSourcesToRequestLocked();
   }
 
   // Outside of lock
-  request(requestSpec);
+  request(requestSpecs);
   return pages;
 }
 
-void ExchangeClient::request(const RequestSpec& requestSpec) {
+void ExchangeClient::request(const std::vector<RequestSpec>& requestSpecs) {
   auto self = shared_from_this();
-  for (auto& source : requestSpec.sources) {
-    auto future = source->request(requestSpec.maxBytes, kDefaultMaxWaitSeconds);
+  for (auto& spec : requestSpecs) {
+    auto future = folly::SemiFuture<ExchangeSource::Response>::makeEmpty();
+    if (spec.maxBytes == 0) {
+      future = spec.source->requestDataSizes(kDefaultMaxWaitSeconds);
+    } else {
+      future = spec.source->request(spec.maxBytes, kDefaultMaxWaitSeconds);
```

Comment:
Reduce max wait for actual data fetching

Reply:
@mbasmanova It's a little tricky to do it in this change as we are restrict ourselves to seconds as minimal units.  I will do it in a few separate changes on both Velox and presto_cpp side to address this.  I will put 1 second here for now.

Reply:
Got it. Thanks.